### PR TITLE
Model classes

### PR DESCRIPTION
**Pull request by Onne164**
Added 9 model classes

### CODE REVIEW

File: src/main/java/ee/thenewyou/personaltrainer/model/Unit.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package ee.thenewyou.personaltrainer.model;
+
+public enum Unit {
```
Comment (StanislavRa) [inline src/main/java/ee/thenewyou/personaltrainer/model/Unit.java:3]: we do not need this class anymore

File: src/main/java/ee/thenewyou/personaltrainer/model/IngredientsSet.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import javax.persistence.*;
+
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+
+@Entity
+@Table (name= "ingredients_Set")
+public class IngredientsSet {
```
Comment (StanislavRa) [inline src/main/java/ee/thenewyou/personaltrainer/model/IngredientsSet.java:13]: we do not need this class anymore

File: src/main/java/ee/thenewyou/personaltrainer/model/ChallengeWeek.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+
+
+@Entity
+@Table(name = "challenge_week")
+
+public class ChallengeWeek {
```
Comment (StanislavRa) [inline src/main/java/ee/thenewyou/personaltrainer/model/ChallengeWeek.java:18]: we do not need this class anymore

File: src/main/java/ee/thenewyou/personaltrainer/model/Challenge.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    @Column(name = "challenge_name")
+    private String challengeName;
+
+    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER, mappedBy="challenge")
+    private List<ChallengeWeek> weeks = new ArrayList<>();
+
+    @Column(name = "start_date")
+    private Date startDate;
+
```
Comment (StanislavRa) [inline src/main/java/ee/thenewyou/personaltrainer/model/Challenge.java:32]: add a new field as weeksQuantity: int

File: src/main/java/ee/thenewyou/personaltrainer/model/ChallengeDay.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    private List<Exercise> exercises = new ArrayList<>();
+
+    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER, mappedBy="meals")
+    private List<Meal> meals = new ArrayList<>();
+
+    @Column(name = "date", nullable = false)
+    private Date date;
+
+    @Column(length = 1, name= "Training_done")
+    private boolean isTrainingDone;
```
Comment (StanislavRa) [inline src/main/java/ee/thenewyou/personaltrainer/model/ChallengeDay.java:34]: we do not need this anymore here

File: src/main/java/ee/thenewyou/personaltrainer/model/ChallengeDay.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Column(length = 1, name= "Nutrition_done")
+    private boolean isNutritionDone;
```
Comment (StanislavRa) [inline src/main/java/ee/thenewyou/personaltrainer/model/ChallengeDay.java:37]: we do not need this anymore here

File: src/main/java/ee/thenewyou/personaltrainer/model/Ingredient.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+
+
+@Entity
+@Table (name= "ingredients")
+public class Ingredient {
```
Comment (StanislavRa) [inline src/main/java/ee/thenewyou/personaltrainer/model/Ingredient.java:14]: we do not need this class anymore

File: src/main/java/ee/thenewyou/personaltrainer/model/ChallengeDay.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+@Entity
+@Table(name = "challenge_day")
+public class ChallengeDay {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "day_id")
+    private long dayId;
+
+    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER, mappedBy="exercises")
```
Comment (StanislavRa) [inline src/main/java/ee/thenewyou/personaltrainer/model/ChallengeDay.java:24]: should be Many-To-Many

File: src/main/java/ee/thenewyou/personaltrainer/model/ChallengeDay.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    private List<Exercise> exercises = new ArrayList<>();
+
+    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER, mappedBy="meals")
```
Comment (StanislavRa) [inline src/main/java/ee/thenewyou/personaltrainer/model/ChallengeDay.java:27]: should be Many-To-Many

File: src/main/java/ee/thenewyou/personaltrainer/model/ChallengeDay.java [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+    private List<Meal> meals = new ArrayList<>();
+
+    @Column(name = "date", nullable = false)
+    private Date date;
```
Comment (StanislavRa) [inline src/main/java/ee/thenewyou/personaltrainer/model/ChallengeDay.java:31]: we do not need this anymore

File: src/main/java/ee/thenewyou/personaltrainer/model/Challenge.java [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+@Entity
+@Table(name = "challenge")
+public class Challenge {
+
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "challenge_id")
+    private long challengeId;
```
[the remark this answers is not in the document]
Comment (Onne164) [reply src/main/java/ee/thenewyou/personaltrainer/model/Challenge.java]: long has been changed to Long in all classes

File: src/main/java/ee/thenewyou/personaltrainer/model/Challenge.java [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+
+    @Column(name = "challenge_name")
+    private String challengeName;
+
+    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER, mappedBy="challenge")
```
[the remark this answers is not in the document]
Comment (Onne164) [reply src/main/java/ee/thenewyou/personaltrainer/model/Challenge.java]: Fetch type eager has been removed from everywhere

File: src/main/java/ee/thenewyou/personaltrainer/model/Challenge.java [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+    private List<ChallengeWeek> weeks = new ArrayList<>();
+
+    @Column(name = "start_date")
+    private Date startDate;
+
+    @Column(name = "end_date")
+    private Date endDate;
+
+    @Column(name = "board")
+    private ChatBoard chatboard;
```
[the remark this answers is not in the document]
Comment (Onne164) [reply src/main/java/ee/thenewyou/personaltrainer/model/Challenge.java]: changed to One-To-One mapping as a comment

File: src/main/java/ee/thenewyou/personaltrainer/model/Challenge.java [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+
+    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER, mappedBy="challenge")
+    private List<ChallengeWeek> weeks = new ArrayList<>();
+
+    @Column(name = "start_date")
+    private Date startDate;
+
+    @Column(name = "end_date")
+    private Date endDate;
+
```
[the remark this answers is not in the document]
Comment (Onne164) [reply src/main/java/ee/thenewyou/personaltrainer/model/Challenge.java:36]: added created_on and updated_on timestamps

File: src/main/java/ee/thenewyou/personaltrainer/model/ChallengeWeek.java [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "week_id")
+    private long weekId;
+
+    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER, mappedBy="challenge_week")
+    private List<ChallengeDay> days = new ArrayList<>();
+
+    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER, mappedBy="ingredients")
+    private List<IngredientsSet> shoppingList = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (Onne164) [reply src/main/java/ee/thenewyou/personaltrainer/model/ChallengeWeek.java]: this feature is deleted

File: src/main/java/ee/thenewyou/personaltrainer/model/ChatBoard.java [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+@Entity
+@Table(name = "chatboard")
+
+public class ChatBoard {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long boardId;
+
+    @ManyToOne
```
[the remark this answers is not in the document]
Comment (Onne164) [reply src/main/java/ee/thenewyou/personaltrainer/model/ChatBoard.java]: changed to One-To-Many

File: src/main/java/ee/thenewyou/personaltrainer/model/Unit.java (hunk 1 above)
Comment (Onne164) [reply src/main/java/ee/thenewyou/personaltrainer/model/Unit.java]: class has been deleted

File: src/main/java/ee/thenewyou/personaltrainer/model/IngredientsSet.java (hunk 2 above)
Comment (Onne164) [reply src/main/java/ee/thenewyou/personaltrainer/model/IngredientsSet.java]: this class has been deleted

File: src/main/java/ee/thenewyou/personaltrainer/model/ChallengeWeek.java (hunk 3 above)
Comment (Onne164) [reply src/main/java/ee/thenewyou/personaltrainer/model/ChallengeWeek.java]: this class has been deleted

File: src/main/java/ee/thenewyou/personaltrainer/model/Challenge.java (hunk 4 above)
Comment (Onne164) [reply src/main/java/ee/thenewyou/personaltrainer/model/Challenge.java:33]: added new field weekQuantity: int

File: src/main/java/ee/thenewyou/personaltrainer/model/ChallengeDay.java (hunk 5 above)
Comment (Onne164) [reply src/main/java/ee/thenewyou/personaltrainer/model/ChallengeDay.java]: deleted

File: src/main/java/ee/thenewyou/personaltrainer/model/ChallengeDay.java (hunk 6 above)
Comment (Onne164) [reply src/main/java/ee/thenewyou/personaltrainer/model/ChallengeDay.java]: deleted

File: src/main/java/ee/thenewyou/personaltrainer/model/Ingredient.java (hunk 7 above)
Comment (Onne164) [reply src/main/java/ee/thenewyou/personaltrainer/model/Ingredient.java]: this class has been deleted

File: src/main/java/ee/thenewyou/personaltrainer/model/ChallengeDay.java (hunk 8 above)
Comment (Onne164) [reply src/main/java/ee/thenewyou/personaltrainer/model/ChallengeDay.java]: changed to Many-To-Many

File: src/main/java/ee/thenewyou/personaltrainer/model/ChallengeDay.java (hunk 9 above)
Comment (Onne164) [reply src/main/java/ee/thenewyou/personaltrainer/model/ChallengeDay.java]: changed to Many-To-Many

File: src/main/java/ee/thenewyou/personaltrainer/model/ChallengeDay.java (hunk 10 above)
Comment (Onne164) [reply src/main/java/ee/thenewyou/personaltrainer/model/ChallengeDay.java]: deleted

**Pull request merged**